bitcoin/src/internal_macros.rs

```diff
@@ -211,3 +211,108 @@ macro_rules! impl_asref_push_bytes {
     };
 }
 pub(crate) use impl_asref_push_bytes;
+
+/// Defines an trait `$trait_name` and implements it for `ty`, used to define extension traits.
+macro_rules! define_extension_trait {
+    // With self, no generics.
+    ($(#[$($trait_doc_attrs:tt)*])* $trait_vis:vis trait $trait_name:ident impl for $ty:ident {
+        $(
+            $(#[$($fn_doc_attrs:tt)*])*
+            fn $fn:ident($slf:ident $(, $param_name:ident: $param_type:ty)*) -> $( $ret:ty )? $body:block
```

Comment:
`->` needs to be inside the following parentheses.

Reply:
ooo, yes. Thanks.